transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -47,6 +49,7 @@
             new IllegalStateException("Channel is unhealthy not offering it back to pool"),
             SimpleChannelPool.class, "releaseAndOffer(...)");
 
+    protected final ConcurrentMap<Channel, Channel> allChannels = PlatformDependent.newConcurrentHashMap();
```

Comment:
Hi @normanmaurer, I'm afraid not because channel pool needs  ``putIfAbsent''. Thanks.